packages/web-app-unzip/src/composables/useUnzipAction.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  FileAction,
+  FileActionOptions,
+  UppyService,
+  useClientService,
+  useLoadingService,
+  useResourcesStore,
+  useService,
+  useUserStore,
+  useWebWorkersStore
+} from '@ownclouders/web-pkg'
+import { computed, unref } from 'vue'
+import { useGettext } from 'vue3-gettext'
+import { Archive } from 'libarchive.js'
+import ArchiveWorker from 'libarchive.js/dist/worker-bundle.js?url'
+import { extractNameWithoutExtension, urlJoin } from '@ownclouders/web-client'
+import { UppyFile } from '@uppy/core'
+import * as uuid from 'uuid'
+
+const SUPPORTED_MIME_TYPES = ['application/zip', 'application/x-tar']
+
+export const useUnzipAction = () => {
+  const { $gettext } = useGettext()
+  const clientService = useClientService()
+  const loadingService = useLoadingService()
+  const { createWorker, terminateWorker } = useWebWorkersStore()
+  const userStore = useUserStore()
+  const resourcesStore = useResourcesStore()
+  const uppyService = useService<UppyService>('$uppyService')
+
+  const createRootFolder = async ({ space, resources }: FileActionOptions) => {
+    const archiveName = extractNameWithoutExtension(resources[0])
+
+    // conflict handling
+    let folderName: string
+    let conflictResolved = false
+    let iterations = 1
+    while (!conflictResolved) {
+      folderName = `${archiveName} (${iterations.toString()})`
+      iterations++
+
+      if (resourcesStore.resources.some(({ name }) => name === folderName)) {
+        continue
+      }
+
+      conflictResolved = true
+    }
+
+    const path = urlJoin(resourcesStore.currentFolder.path, folderName)
+    const resource = await clientService.webdav.createFolder(unref(space), { path })
+    resourcesStore.upsertResource(resource)
+    return resource
+  }
```

Comment:
Would it make sense to create a helper for this in maybe web-client or web-pkg?

Reply:
In the long term, definitely. Ideally we refactor this together with the conflict handling we already have. But this stuff is quite messy, as you know... I don't want to open this can of worms just now.